src/player.cpp

```diff
@@ -938,7 +938,7 @@ void Player::sendPing()
 	}
 
 	if (noPongTime >= noPongKickTime) {
-		if (isConnecting || getTile()->hasFlag(TILESTATE_NOLOGOUT)) {
+		if (isConnecting || getTile()->hasFlag(TILESTATE_NOLOGOUT) || pzLocked) {
```

Comment:
hypothetically, in rare situation when `getTile` returns `nullptr` this may crash

I'd suggest `getTile() && getTile()->hasFlag(TILESTATE_NOLOGOUT)`

Reply:
I will use your suggestion and wait for the others.